app/controllers/application_controller.rb

```diff
@@ -546,6 +546,10 @@ def verify_edit_not_admin
     verify_edit(false)
   end
 
+  def verify_owner
+    raise User::Forbidden, 'Restricted to users with owner permissions.' unless @user.owner(@notebook)
```

Comment:
Should also allow admin here

Reply:
The method in the user model covers admin there already. 
Tested it and it checks out.

```def owner(notebook)
    if notebook != nil
      type = Notebook.find(notebook.id).owner_type
      if ((type == "User" && notebook.owner_id == id) || (type == "Group" && GroupMembership.where(user_id: id, group_id: Group.find(notebook.owner_id)).pluck(:owner)[0]) || admin?)
        return true
      end
    else
      return false
    end
  end```